test/API/titerate.c

```diff
@@ -438,161 +438,162 @@ test_iter_attr(hid_t fapl, hbool_t new_format)
     char     *anames[NATTR]; /* Names of the attributes created */
     iter_info info;          /* Custom iteration information */
     herr_t    ret;           /* Generic return value        */
-#else
     (void)fapl;
     (void)new_format;
-#endif
 
     /* Output message about test being performed */
     MESSAGE(
         5,
         ("Testing Attribute Iteration Functionality - SKIPPED for no due to no iteration restart support\n"));
-#ifndef NO_ITERATION_RESTART
-    HDmemset(&info, 0, sizeof(iter_info));
-
-    /* Create the test file with the datasets */
-    file = H5Fcreate(DATAFILE, H5F_ACC_TRUNC, H5P_DEFAULT, fapl);
-    CHECK(file, FAIL, "H5Fcreate");
-
-    filespace = H5Screate(H5S_SCALAR);
-    CHECK(filespace, FAIL, "H5Screate");
-
-    dataset = H5Dcreate2(file, "Dataset", H5T_NATIVE_INT, filespace, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
-    CHECK(dataset, FAIL, "H5Dcreate2");
-
-    for (i = 0; i < NATTR; i++) {
-        HDsnprintf(name, sizeof(name), "Attribute %02d", i);
-        attribute = H5Acreate2(dataset, name, H5T_NATIVE_INT, filespace, H5P_DEFAULT, H5P_DEFAULT);
-        CHECK(attribute, FAIL, "H5Acreate2");
+    if ((vol_cap_flags_g & H5VL_CAP_FLAG_ITERATE) && (vol_cap_flags_g & H5VL_CAP_FLAG_FILE_BASIC)) {
```

Comment:
This one probably also needs to check for at least H5VL_CAP_FLAG_DATASET_BASIC and H5VL_CAP_FLAG_ATTR_BASIC. We can also remove the part of the message above that says SKIPPED now that this test should get run.

Reply:
Added, and I removed the skipped part.